arviz/tests/base_tests/test_data.py

```diff
@@ -480,6 +480,28 @@ def test_repr_html(self):
         assert escape(idata.__repr__()) in html
         xr.set_options(display_style=display_style)
 
+    def test_add_groups(self):
+        data = np.random.normal(size=(4, 500, 8))
+        idata = from_dict(
+            posterior={"a": data[..., 0], "b": data},
+            sample_stats={"a": data[..., 0], "b": data},
+            observed_data={"b": data[0, 0, :]},
+            posterior_predictive={"a": data[..., 0], "b": data},
+        )
+        with pytest.raises(ValueError):
+            idata.add_groups()
+        idata.add_groups({"prior": {"a": data[..., 0], "b": data}})
+        assert "prior" in idata._groups
```

Comment:
assert both `prior` is in `_groups` and that `prior` is an attribute of `idata`, also a warmup group should also be checked